src/ca/caContext.cpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright - See the COPYRIGHT that is included with this distribution.
+ * pvAccessCPP is distributed subject to a Software License Agreement found
+ * in file LICENSE that is included with this distribution.
+ */
+
+#include <cadef.h>
+#include <pv/pvAccess.h>
+
+#define epicsExportSharedSymbols
+#include "caContext.h"
+
+namespace epics {
+namespace pvAccess {
+namespace ca {
+
+CAContext::CAContext()
+{
+    int result = ca_context_create(ca_enable_preemptive_callback);
```

Comment:
Check if there's already a context, throw logic-error?

Reply:
There seems no need to throw here if the thread is already attached, creating a new CAContext doesn't need to leave us attached to it anyway, so if the thread was already attached to one I restore it before returning. There was a bug in the destructor though.